core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java

```diff
@@ -1264,18 +1270,26 @@ public static String getOncoKbToken()  {
     }
 
     public static String getDownloadControl() {
-        String downloadControlOption = getProperty("skin.hide_download_controls");
-        /*
-            skin.hide_download_controls   return_value
-                    true                    hide
-                    false                   show
-                    data                    data
-                    null/empty              show
-         */
-        switch ((downloadControlOption != null) ? downloadControlOption.trim().toLowerCase() : "false") {
-            case "true" : return "hide";
-            case "data" : return "data";
-            default: return "show";
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if(authentication != null &&
+            StringUtils.isNotEmpty(downloadGroup) &&
+            authentication.getAuthorities().contains(new SimpleGrantedAuthority(downloadGroup))) {
+            return "show";
+        }else {
+            String downloadControlOption = getProperty("skin.hide_download_controls");
+            /*
+                skin.hide_download_controls   return_value
+                        true                    hide
+                        false                   show
+                        data                    data
+                        null/empty              show
+             */
+            switch ((downloadControlOption != null) ? downloadControlOption.trim().toLowerCase() : "false") {
+                case "true" : return "hide";
+                case "data" : return "data";
+                default: return "show";
+            }
```

Comment:
This control logic appears overly complicated. Code should better document itself. Streamline to?
```suggestion
            String downloadControlOption = getProperty("skin.hide_download_controls").trim().toLowerCase();
            switch (downloadControlOption) {
                case "true" : return "hide";
                case "false": return "show";
                case "data" : return "data";
                default: return "show";
            }
```